src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,336 @@
+        - max_bulk_size: The max amount in bytes of a bulk call
+        - anomaly_detection_trend: list of configurations dicts to use for generating data trends
+
+    Returns:
+        - A list of the data that was ingested
+
+    Raises: 
+        - TypeError: Request body is not a list
+        - ValueError: Request body does not come in index, action pairs; an index name does not have an action or vice versa
+    """
+
+    # Validates that inputs are correct
+    ingest_validation(client = client, 
+        data_template = data_template,
+        index_name = index_name,
+        file_provided = file_provided,  
+        mapping = mapping,  
+        number = number, 
+        chunk = chunk, 
+        timestamp = timestamp,
+        minutes = minutes,
+        current_date = current_date,
+        max_bulk_size = max_bulk_size,
+        anomaly_detection_trend = anomaly_detection_trend
+    )
+
+    dataset = []
+
+    # If the user provides their own data
+    if file_provided:
+        dataset = ingest_from_user_data(filename = data_template)
+
+    # If anomalies wanted to be generated
+    elif anomaly_detection_trend:
+        for i in range(number):
+            # Generate random data
+            entry = generate_data(data_template, mapping)
+            
+            # For each feature needing a trend, modify that specific field value to fit a trend
+            for desired_trend in anomaly_detection_trend:
+                if desired_trend["data_trend"] == "AverageTrend":
+                    changed_entry = AverageTrend(  
+                        timestamp = timestamp, 
+                        entry = entry,
+                        feature_trend = desired_trend, 
+                        current_date = current_date
+                    )
+                    entry = changed_entry.generate_data_trend()
+            dataset.append(entry)
+            # Increment current date
+            current_date += timedelta(minutes = minutes)
+    else:
+         # Generates the specified number of documents
+        for i in range(number):
+            entry = generate_data(data_template, mapping)
+            if type(entry) is list:
+                for element in entry:
+                    if type(element) is str:
+                        element = loads(element)
+                    if timestamp:
+                        element[timestamp] = current_date
+                    element = dumps(element)
+                    dataset.append(element)
+            else:
+                entry = generate_data(data_template, mapping)
+                if type(entry) is str:
+                    entry = loads(entry)
+                if timestamp:
+                    entry[timestamp] = current_date
+                entry = dumps(entry)
+                dataset.append(entry)
+            current_date += timedelta(minutes = minutes)
+
+    # Calls BULK API to ingest documents of size "chunk"
+    i = 0
```

Comment:
can we use a good variable name to instead of i? thanks

Reply:
Done, renamed to `days`